DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java

```diff
@@ -835,24 +847,33 @@ public static boolean isDbArray(Class<?> type) {
                         new NodeList<>(expressions[lastArgIndex]));
                 expressionTypes[lastArgIndex] = convertDBArray(expressionTypes[lastArgIndex],
                         parameterizedTypes[lastArgIndex] == null ? null : parameterizedTypes[lastArgIndex][0]);
-                anyExpressionTypesArePrimitive = false;
+                allExpressionTypesArePrimitive = false;
             } else {
                 for (int ei = nArgs - 1; ei < nArgExpressions; ei++) {
                     // iterate over the vararg argument expressions
-                    if (varArgType != expressionTypes[ei] && varArgType.isPrimitive()
-                            && expressionTypes[ei].isPrimitive()) {
+                    if (varArgType == expressionTypes[ei]) {
+                        continue;
+                    } else if (argumentTypes[nArgs - 1] == expressionTypes[ei]) {
```

Comment:
If the LHS is assignable to the RHS, we're already good. And we should do that before the loop.